### PR TITLE
ref(perf): Move out the span transactions table

### PR DESCRIPTION
**Pull request by gggritso**
This one's gross! In short, we had a component called the `SpanTransactionsTable`. It was a transactions table with some conditional logic, to render different columns and fetch different data depending on what module was using it.

The problem? The only module using it is the database module. So, I'm dumping this "dynamic" table, and replacing it with a normal table, the kind we have all over the place. This will make it easier to consolidate the code, and remove a few rarely-used utils like `extractRoute`.

I'm also removing some cruft like the competing `endpoint` and `transaction` parameters that we don't use anymore, lifting up the data loading state, etc.

- Move transactions table into the module
- Lift up data fetching state
- Static column order
- Remove redundant wrapper
- Simplify routing
- Move out table cell renderer
- Export testable component
- Sorting

### CODE REVIEW

File: static/app/views/performance/database/queryTransactionsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  'avg(span.self_time)',
+  'spm()',
+  'time_spent_percentage()',
+] as const;
+
+type ValidSort = Sort & {
+  field: (typeof SORTABLE_FIELDS)[number];
+};
+
+export function isAValidSort(sort: Sort): sort is ValidSort {
```
Comment (0Calories) [inline static/app/views/performance/database/queryTransactionsTable.tsx:73]: neat

File: static/app/views/performance/database/databaseSpanSummaryPage.tsx [hunk 2]
```diff
@@ -89,6 +85,29 @@ export function DatabaseSpanSummaryPage({params}: Props) {
...
+      'spm()',
+      `sum(${SpanMetricsField.SPAN_SELF_TIME})`,
+      `avg(${SpanMetricsField.SPAN_SELF_TIME})`,
+      'time_spent_percentage()',
+      `${SpanFunction.HTTP_ERROR_COUNT}()`,
+    ],
+    sorts: [sort],
+    limit: TRANSACTIONS_TABLE_ROW_COUNT,
+    cursor,
+    referrer: 'api.starfish.span-transaction-metrics',
```
Comment (0Calories) [inline static/app/views/performance/database/databaseSpanSummaryPage.tsx:108]: we should probably rename this referrer eventually

**Review by 0Calories — APPROVED** [review]
Nice refactor! :shipit:

File: static/app/views/performance/database/databaseSpanSummaryPage.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/performance/database/databaseSpanSummaryPage.tsx:108]: Yes, definitely! Cleaning up referrers everywhere is on my list

**Pull request merged**